conwayste/src/ui/ui_errors.rs

```diff
@@ -15,11 +15,33 @@
  *  You should have received a copy of the GNU General Public License
  *  along with libconway.  If not, see <http://www.gnu.org/licenses/>. */
 
+use crate::ggez::GameError;
+
 custom_error! {pub UIError
     InvalidDimensions {reason: String} = "UIError::InvalidDimensions({reason})",
     WidgetNotFound {reason: String} = "UIError::WidgetNotFound({reason})",
     InvalidAction {reason: String} = "UIError::InvalidAction({reason})",
-    ActionRestricted{reason: String} = "UIError::ActionRestricted({reason})"
+    ActionRestricted{reason: String} = "UIError::ActionRestricted({reason})",
+    WidgetIDCollision{reason: String} = "UIError::WidgetIDCollision({reason})",
+    GgezError{reason: String} = "UIError::GgezError({reason})"
```

Comment:
Not sure this is a good idea -- it is probably better to preserve the `GameError` rather than making a new string out of its string message.

Reply:
It was actually unused. Gunk from an attempt at converting between `UIError` and `GameError`s, which was resolved by the code written further below. Good catch!